This is required to enable builds in architectures other than amd64
(x86_64). Currently supports i386/armv6/armv7/aarch64.

The upstream as of now "hardcodes" amd64 (x86_64) as the only build
option.

ARCH_NAME is now set from configure.ac

--- variables.am.orig	2019-03-11 19:42:59 UTC
+++ variables.am
@@ -1,5 +1,3 @@
-ARCH_NAME = arch_x86_64
-
 LANGFLAGS = -std=c99 -fno-strict-aliasing
 
 WARNFLAGS = -pedantic -Wall -Wextra -Werror -Wunreachable-code \
@@ -15,8 +13,10 @@ INCFLAGS += -I$(top_srcdir)/src/include/$(ARCH_NAME)
 
 OFLAGS ?= -O2
 
-ARCHFLAGS = -maes # -mavx2 -mbmi2 #TODO
-ARCHFLAGS += $(XARCHFLAGS)
+ARCHFLAGS = $(XARCHFLAGS)
+if ARCH_X86
+ARCHFLAGS += -maes # -mavx2 -mbmi2 #TODO
+endif
 GENFLAGS = -ffunction-sections -fdata-sections -fvisibility=hidden -fomit-frame-pointer -fPIC
 
 LANGXXFLAGS = -fno-strict-aliasing
